### PR TITLE
go/governance: Support delegator votes

### PR DESCRIPTION
**Pull request by ptrus**

### CODE REVIEW

File: go/consensus/tendermint/apps/governance/governance.go [hunk 1]
```diff
@@ -382,24 +385,95 @@ func (app *governanceApplication) closeProposal(
...
+
+			}
+
+			// Add stake to vote.
+			currentVotes := proposal.Results[vote]
+			if err := currentVotes.Add(escrow); err != nil {
+				return fmt.Errorf("failed to add votes: %w", err)
+			}
+			proposal.Results[vote] = currentVotes
+			// TODO: also add total stake here.
```
Comment (kostko) [inline go/consensus/tendermint/apps/governance/governance.go:475]: What is this TODO about?

**Review by kostko — COMMENTED** [review]
Can you also update the E2E governance workload to generate non-validator / delegator votes?

**Comment by ptrus** [comment]
>  Can you also update the E2E governance workload to generate non-validator / delegator votes?

Yeah, also noticed some other unit tests are missing, and a part for checking the voter has delegations to validators. Moving back to draft.

File: go/consensus/tendermint/apps/governance/governance.go (hunk 1 above)
Comment (ptrus) [reply go/consensus/tendermint/apps/governance/governance.go:475]: Outdated, I thought about moving the total voting stake computation here, but that would be wrong.

**Comment by ptrus** [comment]
> Can you also update the E2E governance workload to generate non-validator / delegator votes?

Added

File: go/consensus/tendermint/apps/governance/governance.go [hunk 2]
```diff
@@ -382,24 +376,95 @@ func (app *governanceApplication) closeProposal(
...
+			proposal.InvalidVotes++
+		}
+	}
+
+	// Finalize the voting results - convert votes in shares into results in stake.
+	proposal.Results = make(map[governance.Vote]quantity.Quantity)
+	for validator, votes := range validatorVoteShares {
+		validatorPool, ok := validatorEntitiesPool[validator]
+		if !ok {
+			panic("should not happen")
```
Comment (kostko) [inline go/consensus/tendermint/apps/governance/governance.go:446]:
```suggestion
        // This should NEVER happen.
			panic("governance: missing validator pool")
```

**Review by kostko — APPROVED** [review]

**Pull request merged**